lib/utils.js

```diff
@@ -95,15 +127,31 @@ const utils = {
         utils.logError('Configuration file missing for this directory, run losant configure to generate this file.');
         return;
       }
-      config.file = file;
+      if (!config.apiUrl) {
+        // old version of the apiUrl
+        config.apiUrl = process.env.LOSANT_API_URL;
+        await writeFile(foundFile, yaml.safeDump(config));
+      }
+      config.file = foundFile;
     }
     return config;
   },
 
   loadConfig: async (file) => {
-    const userConfig = await utils.loadUserConfig(file);
+    let userConfig = await utils.loadUserConfig();
     const appConfig = await utils.loadApplicationConfig(file);
-    return merge(userConfig, appConfig);
+    if (!appConfig) { return {}; }
+    // backwards compatible in case the current applicationConfig does not have an apiUrl
+    const apiKey = appConfig.apiUrl;
+    userConfig = userConfig[apiKey];
```

Comment:
if applicationConfig doesn't have an apiUrl, this is going to end up returning nothing, even if there is userConfig

Reply:
in the function `loadApplicationConfig` it is defaulting apiUrl to process.env.LOSANT_API_URL, so I don't think this will ever happen.

Reply:
okay I've changed it so loadApplicationConfig will try to default apiUrl to process.env.LOSANT_API_URl since that's what they should be setting if they are using any other API outside of 'https://api.losant.com'.

But your right the key could still be undefined...here or in application I could default to `https://api.losant.com` if process.env.LOSANT_API_URl is not defined and/or I could see if they only have one configuration setting in the credentials file to just use that? thoughts?